source/agora/consensus/protocol/Nominator.d

```diff
@@ -466,6 +441,48 @@ extern(D):
             log.info("Rejected invalid envelope: {}", scpPrettify(&envelope));
     }
 
+    /***************************************************************************
+
+        Store the latest SCP sate for restore
+
+    ***************************************************************************/
+
+    public void storeLatestState () @safe nothrow
+    {
+        vector!SCPEnvelope envelopes;
+
+        () @trusted
+        {
+            if (this.scp.empty())
+                return;
+            envelopes = this.scp.getLatestMessagesSend(this.scp.getHighSlotIndex());
```

Comment:
Changed from getCurrentState() to getLatestMessagesSend()
getLatestMessagesSend is returns the latest messages sent for the given slot.

Reply:
The problem is not visible in this version.

Reply:
Just as a note to readers: We discussed this IRL. Stellar does the same thing here, using `getLatestMessagesSend()` instead of `getCurrentState`. It does mean that messages from other nodes are not recovered. However, the node will accept **new** messages from other nodes later when they are sent. It's only its own internal state that it needs to recover - IOW the node's own messages that it gossips.

This may not be very clear to people unless they understand SCP well.